database_test.go

```diff
@@ -552,38 +552,15 @@ func TestDatabaseUpdate(t *testing.T) {
 		t.Fatalf("update 5, database state mismatch:\ngot  %+v\nwant %+v", gotDB, wantDB)
 	}
 
-	// Update 6: api is broken for some unknown reason. Checks the backoff
+	// Update 6: api is broken for some unknown reason. Checks retrial delay.
 	errResponse = errors.New("Something broke")
 	delay, updated = db.Update(context.Background(), mockAPI)
 	if db.err == nil || updated {
 		t.Fatalf("update 6, unexpected update success")
 	}
-	minDelay := baseRetryDelay.Seconds() * float64(1) * float64(1)
-	maxDelay := baseRetryDelay.Seconds() * float64(2) * float64(1)
-	if delay.Seconds() < minDelay || delay.Seconds() > maxDelay {
-		t.Fatalf("update 6, Expected delay %v to be between %v and %v", delay.Seconds(), minDelay, maxDelay)
-	}
-
-	// Update 7: api is still broken, check backoff is larger
```

Comment:
We should have tests somewhere for the exponential backoff. Probably in api_test.go (that test is a bit ugly right now). I also notice that the API doesn't get the 'Now()' from the configuration, which may make it difficult to write time based tests.